src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -58,195 +63,147 @@ Device and XCLBIN
+
+The class constructor ``xrt::device::device(const std::string& bdf)`` also supports opening a device object from a Pcie BDF passed as a string.
 
 .. code:: c++
       :number-lines: 10
            
-           unsigned int dev_index = 0;
-           auto device = xrt::device(dev_index);
-           auto xclbin_uuid = device.load_xclbin("kernel.xclbin");
-       
-The above code block shows
+           auto device = xrt::device("0000:03:00.1");
 
-    - The ``xrt::device`` class's constructor is used to open the device
-    - The member function ``xrt::device::load_xclbin`` is used to load the XCLBIN from the filename. 
-    - The member function ``xrt::device::load_xclbin`` returns the XCLBIN UUID, which is required to open the kernel (refer the Kernel Section). 
 
+The ``xrt::device::get_info()`` is a useful member function to obtain necessary information about a device. Some of the information such as Name, BDF can be used to select a specific device to load an XCLBIN
+
+.. code:: c++
+      :number-lines: 10
+      
+           std::cout << "device name:     " << device.get_info<xrt::info::device::name>() << "\n";
+           std::cout << "device bdf:      " << device.get_info<xrt::info::device::bdf>() << "\n";
 
 Buffers
 -------
 
 Buffers are primarily used to transfer the data between the host and the device. The Buffer related APIs are discussed in the following three subsections
 
-       1. Buffer allocation and deallocation
-       2. Data transfer using Buffers
-       3. Miscellaneous other Buffer APIs
+1. Buffer allocation and deallocation
+2. Data transfer using Buffers
+3. Miscellaneous other Buffer APIs
 
 
 
 1. Buffer allocation and deallocation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-XRT APIs provides API for
+The C++ interface for buffers as below
    
-      - ``xrtBOAlloc``: Allocates a buffer object 4K aligned, the API must be called with appropriate flags. 
-      - ``xrtBOAllocUserPtr``: Allocates a buffer object using pointer provided by the user. The user pointer must be aligned to 4K boundary. 
-      - ``xrtBOFree``: Deallocates the allocated buffer. 
+The class constructor ``xrt::bo`` is mainly used to allocates a buffer object 4K aligned. By default, a regular buffer is created (optionally the user can create other types of buffers by providing a flag). 
 
-.. code:: c
+.. code:: c++
       :number-lines: 15
            
-           xrtMemoryGroup bank_grp_idx_0 = xrtKernelArgGroupId(kernel, 0);
-           xrtMemoryGroup bank_grp_idx_1 = xrtKernelArgGroupId(kernel, 1);
-
-           xrtBufferHandle input_buffer = xrtBOAlloc(device, buffer_size_in_bytes, XRT_BO_FLAGS_NONE, bank_grp_idx_0);
-           xrtBufferHandle output_buffer = xrtBOAlloc(device, buffer_size_in_bytes, XRT_BO_FLAGS_NONE, bank_grp_idx_1);
+           auto bank_grp_idx_0 = kernel.group_id(0);
+           auto bank_grp_idx_1 = kernel.group_id(1);
```

Comment:
Is it explained that the index used for group_id(index) is the argument index to the kernel?
In some ways line 123 and 124 would be clearer if 
```
auto bank_grp_arg0 = kernel.group_id(0);
auto bank_grp_arg1 = kernel.group_id(1);

```
Actually below I see that group_id is indeed explained.  Still you may want a comment here and more descriptive variable names?

Reply:
done the variable change and added code comment for clarity